File: solutions/clojure/164-non-overlapping-intervals.clj
;; Problem 164: Non-overlapping Intervals (LeetCode 435)
;; Difficulty: Med
;; Language: Clojure
;; 
(defn erase-overlap-intervals [intervals]
  (let [sorted (sort-by second intervals)]
    (first
      (reduce (fn [[cnt end] [s e]]
                (if (< s end) [(inc cnt) end] [cnt e]))
              [0 Long/MIN_VALUE] sorted))))

File: solutions/typescript/164-non-overlapping-intervals.ts
/**
 * Problem 164: Non-overlapping Intervals (LeetCode 435)
 * Difficulty: Med
 * Language: TypeScript
 */
function eraseOverlapIntervals(intervals: number[][]): number {
  const sorted = [...intervals].sort((a, b) => a[1] - b[1]);
  return sorted.reduce(
    ([count, end], [s, e]) =>
      s < end ? [count + 1, end] as [number, number]
              : [count, e] as [number, number],
    [0, -Infinity] as [number, number]
  )[0];
}
